### DISCUSSION TITLE
Weighting algorithems

### DISCUSSION THREAD

**Discussion by tomershukhman** [dailydotdev/daily#896] [Feature Requests]
Hi,

For the sake of explaning my though/request let's look at a simple ranking system:
if a user has a tag it is assinged the value 1, 
if he doesn't 0
if he marked at some point "not intersted in <some_tag>" then -1

It would be nice if the suggestion feed could "learn" about the prefrences of a user.
Say a user have the tags "AI","OpenAI","Python"  

Currently (as far as I understand) a post like "The business side of Chat GPT" could pop up in the feed.
The posts tags could be: "OpenAI","AI","Business model","Company analysis" etc
Say at one point I marked not intrested in "Business analysis" it also safe to assume that I"m not in "company analysis" and there for assign it -1

After applying the ranking the average rank of this post is [1+(-1)+(-1)]/3<0 and therefor would not be suggested.

Of course this example is just for the sake of explanation and a much more sophisticated ranking system is desired.

Love daily.dev keep up the good word!

**Comment by idoshamun** [comment]
Hi @tomershukhman,

We released not so long ago a new [algorithm version](https://app.daily.dev/posts/1IuFA6CEl) that includes a bit more complex calculation than this 😉 
It isn't applied to 100% of your feed card as we want to motivate you to explore new topics as well.
My best suggestion would be to upvote content you like so our algorithm learn better your likings.
I hope it makes sense 🙏
